lib/puppet/pops/functions/dispatch.rb

```diff
@@ -72,14 +72,17 @@ def weave(scope, args)
       new_args = []
       @weaving.each do |knit|
         if knit.is_a?(Array)
-          injection_data = @injections[knit[0]]
+          injection_name = @injections[knit[0]]
           new_args <<
-            case injection_data[3]
-            when :dispatcher_internal
-              # currently only supports :scope injection
+            case injection_name
+            when :scope
               scope
+            when :pal_compiler
+              Puppet.lookup(:pal_compiler)
+            when :pal_script_compiler
+              Puppet.lookup(:pal_script_compiler)
```

Comment:
I think the logic above needs to both lookup `:pal_compiler` and that the "pal" version also asserts that it is a {{PalScriptCompiler}}. Also, if not finding a `:pal_compiler`, the default should be to create an instance of `Pal::Compiler` with the `scope.compiler` as an argument. This way, a function that only wants a generic compiler (to find functions etc.) can always get that even if call did not come from via Pal.